src/models/experts/asyncActions.ts

```diff
@@ -47,13 +47,69 @@ export const fetchExperts = createAsyncThunk(
       return {
         expertIds: appendIds,
         experts,
-        meta: {
-          pageNumber: response.data.number,
-          totalPages: response.data.totalPages,
-          totalElements: response.data.totalElements,
-          isLastPage: response.data.last,
-          appendExperts,
+        totalPages: response.data.totalPages,
+        totalElements: response.data.totalElements,
+        isLastPage: response.data.last,
+        appendExperts,
+      };
+    } catch (err) {
+      return rejectWithValue(err.response?.data);
+    }
+  },
+);
+
+export const fetchExpertsAutorsList = createAsyncThunk(
+  'experts/loadExpertsAuthorsList',
+  async (options: IExpertsAutorsListOptions, { getState, rejectWithValue }) => {
+    try {
+      const {
+        experts: {
+          data,
+          meta: {
+            size,
+            textFields: { author },
+            sort: { order, sortBy },
+          },
+        },
+      } = getState() as any;
+
+      const isSearchByAuthorexists = (arg: string) => {
+        if (!arg) {
```

Comment:
separate question: why to do check for `arg`? I understand it as "if we search for specific author we don't pass sorting argument". But Is it really necessary? If there is just single match, then it does not matter if we sort or not. And if there are more than 1 match found, we definitely better to have results ordered by gives field. Or does backend reject having both filtering and sorting?

Reply:
it is a temporary solution because of the bug on the server side while having both filtering and sorting

Reply:
I don't insist, but maybe it does not really make sense to have separate function after logic has been simplified; as a bonus we could put comment on that in single place(with separate function it's harder to find right place for a comment):

```
sort: author ? undefined : `${sortBy},${order}` // backend crashes if both filtering and sorting are passed, see <link to backend issue>
```

Reply:
Done